### DISCUSSION TITLE
Build custom AirSim env for Linux without GUI

### DISCUSSION THREAD

**Discussion by rhalaly** [microsoft/AirSim#4244] [Support Q&A] [answered]
I have built [custom AirSim env](https://github.com/microsoft/AirSim/wiki/build_FSTDriverless_windows) in my Windows machine. I want to run some experiments on Azure Batch with a Linux machine.

I followed the tutorial and got an `exe` file as output, but I want Linux artifacts that will run without any GUI.

I saw [how to build a custom AirSim env](https://github.com/microsoft/AirSim/wiki/build_FSTDriverless_windows) and [AirSim Development Environment on Azure](https://microsoft.github.io/AirSim/azure/), but still not sure how to do that 🙂. 

Is there any tutorial on how to build custom env for Azure Bath with Linux (without GUI and GPU)?

**Comment by zimmy87** [comment, marked as answer]
From looking [here](https://microsoft.github.io/AirSim/docker_ubuntu/#running-an-unreal-binary-inside-a-docker-container), there's a `-headless` option in the run_airsim_image_binary.sh used with docker. If we look at what this option is doing [here](https://github.com/microsoft/AirSim/blob/master/docker/run_airsim_image_binary.sh#L67), it's setting `SDL_VIDEODRIVER = 'offscreen'`. I'm not very familiar with Azure Batch, but you should be able to run AirSim with no GUI on Linux using that environment variable.
